scapy/fields.py

```diff
@@ -201,60 +245,73 @@ class Emph(object):
     __slots__ = ["fld"]
 
     def __init__(self, fld):
+        # type: (Any) -> None
         self.fld = fld
 
     def __getattr__(self, attr):
+        # type: (str) -> Any
         return getattr(self.fld, attr)
 
     def __eq__(self, other):
-        return self.fld == other
+        # type: (Any) -> bool
```

Comment:
No you can't for `__eq__`, `__new__`, `__add__`.... or anything that extends the object class. It makes sense, you can always try to use `==` on another type, so it should remain `Any`. I'll mark as resolved similar cases.

Reply:
I am sad =/

Reply:
Is it a limitation of the Python typing system? With other typed languages, we will be able to restrict the type of `other`. That sounds strange.

Reply:
Here's what mypy returns:

```
mypy run-test: commands[0] | python .config/mypy/mypy_check.py
scapy\fields.py:2541: error: Argument 1 of "__eq__" is incompatible with supertype "object"; supertype defines the argument type as "object"
scapy\fields.py:2541: note: This violates the Liskov substitution principle
scapy\fields.py:2541: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
scapy\fields.py:2541: note: It is recommended for "__eq__" to work with arbitrary objects, for example:
scapy\fields.py:2541: note:     def __eq__(self, other: object) -> bool:
scapy\fields.py:2541: note:         if not isinstance(other, FlagValue):
scapy\fields.py:2541: note:             return NotImplemented
scapy\fields.py:2541: note:         return <logic to compare two FlagValue instances>
```

Reply:
I did some experiments and it seems that there is a `--strict-equality` mypy flag that could help detect things like `Field(...) == 3`. According to mypy documentation and the error message above, we should type ours `__eq__()` and `__ne__()` like:
```
def __eq__(self, other: object) -> bool:
    if not isinstance(other, FlagValue):
       return NotImplemented
    return <logic to compare two FlagValue instances>
```https://stackoverflow.com/questions/37557411/why-does-defining-the-argument-types-for-eq-throw-a-mypy-type-error

I will understand that you don't want to add the `if` block, but could you change `Any` to `object`?

Reply:
@gpotter2 did you try this?

Reply:
Except we don't want to do that, since we want to have
```
>>> FlagValue(1, ["a", "b", "c"])
<Flag 1 (a)>
>>> FlagValue(1, ["a", "b", "c"]) == "a"
True
>>> FlagValue(1, ["a", "b", "c"]) == 1
True
```

Reply:
I really believe that we want to enforce strict type checking. Otherwise, this makes typing quite unhelpful considering the work involved in annotating Scapy. The FlagValue is a valid counter example, but we should handle it separately.

I am OK with merging this PR, as soon as we move this stricter type checking discussions to a dedicated issue =)

Reply:
Sounds good, let's do that in a future PR. See https://github.com/secdev/scapy/issues/2823